estate/models/estate_property_offer.py

```diff
@@ -44,10 +49,35 @@ def action_confirm(self):
         accepted_offer = self.property_id.offer_ids.filtered(lambda o : o.status == 'accepted')
         if accepted_offer:
             raise exceptions.UserError("More than one offer cannot be accepted")
+        if self.property_id.expected_price and self.price < self.property_id.expected_price * 0.9:
+            raise exceptions.UserError("The selling price must be at least 90% of the expected price! You must reduce the expected price if you want to accept this offer'")
         self.status = "accepted"
         self.property_id.selling_price = self.price
         self.property_id.buyer_id = self.buyer_id
 
     def action_deny(self):
         for record in self:
             record.status = "refused"
+
+    _sql_constraints = [
+        (
+            "check_price",
+            "CHECK(price > 0)",
+            "Offer price must be strictly positive",
+        )
+    ]
+
+    @api.model
+    def create(self, vals):
+        property_obj = self.env['property'].browse(vals['property_id'])
+        property_obj.state = 'offer_received'
```

Comment:
This should not be here, what if we reach the raise?

If we reach the raise then the offer is not created, therefore we shouldn't move the state to 'offer received'

Could you change it to be after the if in the raise please?